lib/service.js

```diff
@@ -221,6 +221,21 @@ AWS.Service = inherit({
     var request = new AWS.Request(this, operation, params);
     this.addAllRequestListeners(request);
     this.attachMonitoringEmitter(request);
+    if (this.api.errorCodeMapping) {
+      var errorCodeMapping = this.api.errorCodeMapping;
+      request.on(
+        'error',
+        function (err, response) {
+          if (response.error && response.error.code) {
+            var mapping = errorCodeMapping[response.error.code];
+            if (mapping) {
+              response.error.code = mapping.code;
+            }
+          }
+        },
+        true
+      );
+    }
```

Comment:
Instead of adding it here in the `makeRequest()`, the logic should be moved to [event_listeners.js](https://github.com/aws/aws-sdk-js/blob/master/lib/event_listeners.js) if `errorCodeMapping` trait will be a cross-service trait.

Reply:
Actually, lib/service.js is common for all services. Isn't it?